.github/workflows/release.yml

```diff
@@ -1,61 +1,84 @@
-# Modified Release github action. Replace the current with this when ready.
+# This is the workflow for building release artifacts.
+# Optionally, creating a release with those artifacts is possible by pushing a tag.
 ---
-name: Release New
+name: Release
 
 on:
+  # Allow running this workflow manually from the Actions tab
+  workflow_dispatch:
+  pull_request:
   push:
+    branches:
+      - master
+      - development
     tags:
       - '*'
 
-  # Allows you to run this workflow manually from the Actions tab
-  workflow_dispatch:
-
 jobs:
-  release:
+  Build-Release-Artifacts:
+    name: Build the release artifacts
     runs-on: ubuntu-latest
     steps:
-      - run: |
+      - name: Install host dependencies
+        run: |
           sudo add-apt-repository -y ppa:dysfunctionalprogramming/minisign
           sudo apt update
           sudo apt install -yq build-essential zip minisign
-      - uses: actions/checkout@master
-      - uses: actions-rs/toolchain@v1
+
+      - name: Checkout the repo
+        uses: actions/checkout@master
+
+      - name: Install Rust toolchain
+        uses: actions-rs/toolchain@v1
         with:
           toolchain: stable
-      - name: Build Linux
+
+      # TODO: determine if it is possible to break out the build into multiple targets, to enable parallel job execution
+      - name: Build Distributions
         run: cargo xtask dist
```

Comment:
The way it is now...the builds take over 22 minutes to complete...most of it on this step. If we split the builds out, it is possible to parallelize them with multiple jobs...bringing the time down to more like 8-10 minutes. I can do this here or in a separate issue...depending on a larger conversation about the best way to package and release the artifacts for user consumption. I've already talked with @furi0us333 about this and I believe he plans to have a larger discussion with @louislang

Reply:
I know @andreaphylum implemented this xtask, so I'll defer to him on it. But parallelizing builds is always a plus imo.

Reply:
The xtask is really just [Rust code](https://github.com/phylum-dev/cli/blob/development/xtask/src/main.rs#L60) executed sequentially. We could definitely break it down in more subcommands, ideally one per parallelizable branch. Other than that, if we have a way of precompiling the Docker images and pulling them from somewhere instead of compiling them as a build step, I assume we could further reduce the CPU time. Unfortunately I'm not too well versed with GitHub actions, so not sure if that is feasible, but it would be a pretty big win in that case.

Reply:
Created #188 as a new issue to track this task. Resolving this conversation...